specification/configuration.md

```diff
@@ -130,6 +136,27 @@ beyond the OpenTelemetry specification exist.
   - Distribution MUST default to `otlp`
   - Distribution MUST offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`
 
+### Real User Monitoring Libraries
+
+Real User Monitoring (RUM) instrumentation libraries cannot use environment variables for configuration.
+Instead, they MUST expose a `SplunkRum` object/class/namespace (depending on the language used) that allows
+setting the following properties (in a language-specific way: Java may use builders, Swift can use named
+parameters with default values, JavaScript can use objects, etc.):
+
+| Configuration property (default value) | Description                                                                                                                                                                                                 |
+| -------------------------------------- | -----------                                                                                                                                                                                                 |
+| `beaconUrl` ()                         | RUM beacon URL, ex. `https://rum-ingest.<realm>.signalfx.com/v1/rum`. [1]                                                                                                                                   |
+| `rumAccessToken` ()                    | RUM authentication token. [1]                                                                                                                                                                               |
+| `applicationName` ()                   | Instrumented application name. [1]                                                                                                                                                                          |
+| `globalAttributes` ()                  | OpenTelemetry [Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/common.md#attributes) that will be added to every span produced by the RUM library. |
+
```

Comment:
We should also enforce that an `environmnt` config option is available.

Reply:
yes, I forgot this was something that we should add to the android library.

Reply:
👍 

Should we use `deploymentEnvironment` option (and `deployment.environment` attribute) to match otel conventions?

Currently just `environment` is used: https://github.com/signalfx/splunk-otel-js-web/blob/main/src/index.ts#L265

Reply:
Alright, but if something is clearly just an attribute (that has a semantic meaning in our backend, but doesn't have a semantic meaning in the instrumentation layer), then shouldn't we just say "add it to the global attributes"? It sort of applies similarly to application name.

Reply:
I think there's value in adding separate config options for the most commonly used attributes - it eases the setup.

We're already doing that in otel sdk: there's now `OTEL_SERVICE_NAME=...` option supported even though you could use `OTEL_RESOURCE_ATTRIBUTES=service.name=...` to define exactly the same thing.

> that has a semantic meaning in our backend, but doesn't have a semantic meaning in the instrumentation layer

It has meaning for RUM as a whole though